iothub/device/src/Message.cs

```diff
@@ -85,9 +83,12 @@ internal Message(Stream stream, bool ownStream)
         /// + {'-', ':', '/', '\', '.', '+', '%', '_', '#', '*', '?', '!', '(', ')', ',', '=', '@', ';', '$', '''}.
         /// Non-alphanumeric characters are from URN RFC.
         /// </summary>
+        /// <remarks>
+        /// If this value is not supplied by the user, the device client will set this to a new GUID.
+        /// </remarks>
         public string MessageId
         {
-            get => GetSystemProperty<string>(MessageSystemPropertyNames.MessageId);
+            get => GetSystemProperty<string>(MessageSystemPropertyNames.MessageId) ?? Guid.NewGuid().ToString();
```

Comment:
This is a change in behavior. Currently the MessageId will be null, if unspecified by the user. This however, makes tracing the message very difficult.